Домашнее задание спринт 12

NekhoroshkinaElena opened the pull request.

NekhoroshkinaElena committed:
init commit

Charmandik commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -76,6 +76,57 @@
+                        app:reverseLayout="true" />
+
+                    <Button
+                        android:id="@+id/buttonClearHistory"
```
Для более простого использования различных классов view рекомендую использовать следующий трюк: К имени переменной добавлять префикс в виде "аббревиатуры" класса, к которой она относится, для TextView - tv, ImageView - iv, RecyclerView - rv и т.д.. Например, у тебя есть RecyclerView для треков, в этом случае будет rvTracks

Charmandik commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -46,45 +51,89 @@ class SearchActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_search)
 
         toolbar = findViewById(R.id.toolbar_search)
```
Попробуй применить подход View Binding. ViewBinding позволяет проще писать код для взаимодействия с версткой позволяет избежать этого процесса с инициализацией каждой view по отдельности. Более подробно можно почитать в статье https://habr.com/ru/articles/467295/

Charmandik commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -94,15 +143,19 @@ class SearchActivity : AppCompatActivity() {
+                searchHistoryGroup.visibility =
+                    if (searchField.hasFocus() && s?.isEmpty() == true && searchHistory.getSearchHistory()
+                            .isNotEmpty()
+                    ) View.VISIBLE else View.GONE
```
Для управления видимостью можно использовать свойство isVisible, устанавливая значение false для View.GONE и true для View.VISIBLE
Например, если нам нужно установить View.VISIBLE:
myViewName.isVisible = true
Или если нужно установить View.GONE, то : 
myViewName.isVisible = false

Возможно, тебе больше понравится такой подход

Charmandik commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -46,45 +51,89 @@ class SearchActivity : AppCompatActivity() {
+            buttonUpdate.visibility = View.GONE
         }
 
         val textWatcher = object : TextWatcher {
```
Функцию расширения doOnTextChanged можно использовать вместо класса TextWatcher для отслеживания изменений текста в EditText. Вот пример использования:

editText.doOnTextChanged { text, start, before, count ->
    // Здесь можно выполнять действия при изменении текста    // text - новый текст
    // start - начальная позиция изменения    // before - количество символов, удаленных из текста
    // count - количество добавленных символов в текст}

Метод doOnTextChanged принимает лямбда-выражение, которое будет вызываться при каждом изменении текста. Внутри этого выражения можно выполнять необходимые действия в соответствии с изменениями текста.

NekhoroshkinaElena committed:
refactor: убрать неиспользуемый код в классе SearchActivity.kt
refactor: изменить названия переменных
replace:  применить подход View Binding, вместо инициализации каждой view по отдельности
